core-api/src/main/java/org/silverpeas/core/variables/VariableScheduledValue.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2000 - 2018 Silverpeas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception.  You should have received a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "https://www.silverpeas.org/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.core.variables;
+
+import org.silverpeas.core.date.Period;
+import org.silverpeas.core.persistence.Transaction;
+import org.silverpeas.core.persistence.datasource.model.identifier.UuidIdentifier;
+import org.silverpeas.core.persistence.datasource.model.jpa.SilverpeasJpaEntity;
+import org.silverpeas.core.security.Securable;
+import org.silverpeas.core.util.WebEncodeHelper;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.silverpeas.core.date.TemporalConverter.asLocalDate;
+
+/**
+ * A scheduled value of a given {@link Variable}. Such a value is said scheduled because it is
+ * scheduled to be enabled during a given period of time; during this period, the related variable
+ * is then valued with this value. Don't forget that a variable can have then at the same time
+ * several possible values. In such a case, only the first obtained one will be valid (the first
+ * obtained one isn't necessary the first defined one; in that case we don't ensure any coherent
+ * behavior).
+ */
+@Entity
+@Table(name = "sb_variables_value")
+public class VariableScheduledValue
+    extends SilverpeasJpaEntity<VariableScheduledValue, UuidIdentifier> implements Securable {
+
+  @Column(nullable = false)
+  private String value;
+
+  @Column
+  private LocalDate startDate;
+
+  @Column
+  private LocalDate endDate;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this value-based field transient so it is not included in the serialization of this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3437)